language/stdlib/modules/libra_system_2.move

```diff
@@ -240,18 +242,16 @@ module LibraSystem2 {
         if (i >= size) {
             return false
         };
-        let validator_info = Vector::borrow(validators, i);
+        let validator_info = Vector::borrow_mut(validators, i);
         let new_validator_config = ValidatorConfig2::get_config(validator_info.addr);
         // check if information is the same
-        let config_ref = &mut Vector::borrow_mut(validators, i).config;
-        if (config_ref == &new_validator_config) {
-            return false
-        };
+        let config_ref = &mut validator_info.config;
+        let configs_changed = (config_ref != &new_validator_config);
         *config_ref = *&new_validator_config;
-        true
+        configs_changed
```

Comment:
isn't this change technically more expensive?

Reply:
You are absolutely right! Sorry - it's an artifact of some refactoring attempt that I had to roll back.